File: benchmark-csharp/Controllers/MathController.cs
using Microsoft.AspNetCore.Mvc;

namespace benchmark_csharp.Controllers;

[ApiController]
[Route("math")]
public class MathController : ControllerBase
{
    [HttpGet("fibonacci")]
    public IActionResult Fibonacci([FromQuery] int n = 30)
    {
        return Content(Fib(n).ToString(), "text/plain");
    }

    [HttpGet("fibonacci-iter")]
    public IActionResult FibonacciIter([FromQuery] int n = 30)
    {
        if (n <= 1) return Ok(n);
        long a = 0, b = 1;
        for (int i = 2; i <= n; i++)
        {
            long temp = a + b;
            a = b;
            b = temp;
        }
        return Content(b.ToString(), "text/plain");
    }

    [HttpGet("fibonacci/error")]
    public IActionResult FibonacciError([FromQuery] int n = 30)
    {
        return Content(FibWithError(n).ToString(), "text/plain");
    }

    [HttpPost("matrix/int")]
    public IActionResult MultiplyInt([FromBody] MatrixRequestInt request)
    {
        int[][] A = request.a;
        int[][] B = request.b;
        int n = A.Length, m = A[0].Length, k = B[0].Length;
        int[][] result = new int[n][];
        for (int i = 0; i < n; i++)
        {
            result[i] = new int[k];
            for (int j = 0; j < k; j++)
                for (int l = 0; l < m; l++)
                    result[i][j] += A[i][l] * B[l][j];
        }
        return Ok(result);
    }

    [HttpPost("matrix/float")]
    public IActionResult MultiplyFloat([FromBody] MatrixRequestFloat request)
    {
        double[][] A = request.a;
        double[][] B = request.b;
        int n = A.Length, m = A[0].Length, k = B[0].Length;
        double[][] result = new double[n][];
        for (int i = 0; i < n; i++)
        {
            result[i] = new double[k];
            for (int j = 0; j < k; j++)
                for (int l = 0; l < m; l++)
                    result[i][j] += A[i][l] * B[l][j];
        }
        return Ok(result);
    }

    private long Fib(int n)
    {
        if (n <= 1) return n;
        return Fib(n - 1) + Fib(n - 2);
    }

    private long FibWithError(int n)
    {
        if (n == 1)
            throw new Exception("Error while processing");
        if (n <= 0) return 0;
        return FibWithError(n - 1) + FibWithError(n - 2);
    }

    public class MatrixRequestInt
    {
        public int[][] a { get; set; }
        public int[][] b { get; set; }
    }

    public class MatrixRequestFloat
    {
        public double[][] a { get; set; }
        public double[][] b { get; set; }
    }
}

File: benchmark-php/app/Http/Controllers/MathController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;

class MathController extends Controller
{
    public function fibonacci(Request $request)
    {
        $n = (int)$request->query('n', 30);
        return response((string)$this->fib($n), 200)->header("Content-Type", "text/plain");
    }

    public function fibonacciIter(Request $request)
    {
        $n = (int)$request->query('n', 30);
        if ($n <= 1) return response((string)$n, 200)->header("Content-Type", "text/plain");

        $a = 0;
        $b = 1;
        for ($i = 2; $i <= $n; $i++) {
            $temp = $a + $b;
            $a = $b;
            $b = $temp;
        }

        return response((string)$b, 200)->header("Content-Type", "text/plain");
    }

    public function fibonacciError(Request $request)
    {
        $n = (int)$request->query('n', 30);
        return response((string)$this->fibWithError($n), 200)->header("Content-Type", "text/plain");
    }

    private function fib($n)
    {
        if ($n <= 1) return $n;
        return $this->fib($n - 1) + $this->fib($n - 2);
    }

    private function fibWithError($n)
    {
        if ($n === 1) {
            abort(500, 'Error while processing');
        }
        if ($n <= 0) return 0;
        return $this->fibWithError($n - 1) + $this->fibWithError($n - 2);
    }

    public function multiplyInt(Request $request)
    {
        $A = $request->input('a');
        $B = $request->input('b');
        $n = count($A);
        $m = count($A[0]);
        $k = count($B[0]);
        $result = [];

        for ($i = 0; $i < $n; $i++) {
            $result[$i] = [];
            for ($j = 0; $j < $k; $j++) {
                $sum = 0;
                for ($l = 0; $l < $m; $l++) {
                    $sum += $A[$i][$l] * $B[$l][$j];
                }
                $result[$i][$j] = $sum;
            }
        }

        return response()->json($result);
    }

    public function multiplyFloat(Request $request)
    {
        $A = $request->input('a');
        $B = $request->input('b');
        $n = count($A);
        $m = count($A[0]);
        $k = count($B[0]);
        $result = [];

        for ($i = 0; $i < $n; $i++) {
            $result[$i] = [];
            for ($j = 0; $j < $k; $j++) {
                $sum = 0.0;
                for ($l = 0; $l < $m; $l++) {
                    $sum += (float)$A[$i][$l] * (float)$B[$l][$j];
                }
                $result[$i][$j] = $sum;
            }
        }

        return response()->json($result);
    }
}
